OpenRA.Mods.Common/Scripting/Properties/CombatProperties.cs

```
@@ -96,7 +96,11 @@ public void Attack(Actor targetActor, bool allowMove = true, bool forceAttack =
 			if (!targetActor.Info.HasTraitInfo<FrozenUnderFogInfo>() && !Self.Owner.CanTargetActor(targetActor))
 				Log.Write("lua", "{1} is not revealed for player {0}!", Self.Owner, targetActor);
 
-			attackBase.AttackTarget(target, true, allowMove, forceAttack);
+			var attack = attackBase.FirstOrDefault(x => !x.IsTraitDisabled && x.HasAnyValidWeapons(target));
+			if (attack == null)
+				throw new LuaException("Actor {0} was ordered to attack {1} but has no active Attack* trait to do so!.".F(Self, targetActor));
```

Should this silently fail in that case? Trying to attack in such a scenario also sounds bogus.